### PR TITLE
Max/Project1-CatPinball-Prototype

### PR DESCRIPTION
**Pull request by pecom291**
## Summary

- This is the prototype for the Arcade game for the first project of the semester which is a pinball game around cats! The current PR is now just about complete, The game itself with visuals and mechanics are all working. The overall concept for this was to have a pinball game where you play as a cat swatting a ball of yarn trying to collect treats. I designed it so that you have to click using your mouse the left or right side of the screen in order for the bumpers to move as I felt it added a bit to the game play and makes you feel a bit more involved.

## Details

- One of the first things I felt necessary to add was to make the paddles into cat paws. The scoreboard as well as the background will be primarily cats so the player already is in a happy space opening the game. Most of if not all of the necessary scripting is done but not currently being used, the goal was to roughly outline the game with placeholders and such while doing the coding and other aspects then to fix the backgrounds and everything else. I started making this from actually looking at an old project that I had that was also a 2D game. This helped me a lot in scripting and from there I proceeded to start fixing the bumpers and setting up the rest of the game just in a rough layout. After this I went ahead and threw a bunch of scripts together to try my best to get these pieces to collide and interact with one another which there are a lot of bugs but completely manageable.

- For my Updates I changed around the entirety and fixed the bouncers. There were a few minor tweaks and changes to a few of the scripts to change the times, spawn locations of the ball, and set the basic counter for score to "00" for eyes. All of the assets were drawn by me except for the paws used as paddles. The sound manager was also implemented into the game fully allowing the ball to hit things and make noise, treat breaking makes a crunch, and a kitty meows when you hit the carrier bouncer. 

## Visuals

Citball

## How to Test

- Controls: Left click the LEFT half of the screen to flip left bumper, Left click RIGHT half of the screen to flip right bumper. This game was entirely designed for mobile so switching the game to simulate a pixel 5 will fix almost every single issue.

- To play this game is extremely straightforward. Your left click begins the game, and clicking on the left or right side of the screen with the left mouse button with make the paddle/bumper on that side move. The goal is to hit the ball and to collect the treats above. There is a minor time limit in the game as well that is made so if the treats or ball randomly gets stuck you wont be stuck in a game.

- There is only one known glitch I know of at the moment. The treats like to switch between the old image for them (a red diamond) and the new one of treats. This doesn't effect game play at all I just genuinely ran out of time.

## How to Review

- I would like feedback on how the game plays feels and how the content feels as a whole. What do you think needs improvement? Is the concept good? Are there any little things for me to add that you feel could improve it overall?
- I also want to know if you think there are any betters ways for to have done anything, do you think the ball should be bigger, the spawn points in different places?
- I would also like for indicators of where you think bouncers should be places and how might I revise the layout to possibly make it a little more complex?

### CODE REVIEW

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/Ball.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using UnityEngine;
+
+public class Ball : MonoBehaviour
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/Ball.cs:3]: Namespace

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/Bouncer.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using UnityEngine;
+
+public class Bouncer : MonoBehaviour
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/Bouncer.cs:3]: namespace

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/Diamond.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class Diamond : MonoBehaviour
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/Diamond.cs:4]: namespace

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/Diamond.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        timeToEnd = 10.0f;
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+        if (timeToEnd > 0)
+        {
+            timeToEnd -= Time.deltaTime;
+            Loader.fillAmount = 1 - timeToEnd / 10.0f;
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/Diamond.cs:32]: a classic creative coding pattern 👍

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/FlippersBehaviour.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
+﻿using UnityEngine;
+
+public class FlippersBehaviour : MonoBehaviour
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/FlippersBehaviour.cs:3]: namespace

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/FlippersBehaviour.cs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+{
+
+    [SerializeField]
+    private float tourque;
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/FlippersBehaviour.cs:7]: `torque` spelling

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/FlippersBehaviour.cs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        width = Screen.width / 2.0f;
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+
+        #region UNITY_EDITOR
+
+#if UNITY_EDITOR
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/FlippersBehaviour.cs:29]: Where did you learn this?

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/SoundManager.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SoundManager : MonoBehaviour
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/SoundManager.cs:5]: Namespace

File: Assets/MaxPecora/Assets/Pinball Asset/Scripts/UIManager.cs [hunk 9]
```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class UIManager : MonoBehaviour
```
Comment (jasonzli) [inline Assets/MaxPecora/Assets/Pinball Asset/Scripts/UIManager.cs:6]: namespace

**Review by jasonzli — CHANGES_REQUESTED** [review]
Looks interesting and certainly engineered. I'd like to know where the old project came from. Is it entirely your work? Did you learn this from another source?

**Review by MarioBrnjac — APPROVED** [review]
The way the ball reacts to getting hit is very smooth. I think that right now, it can sometimes move too fast after getting hit and giving it some drag so that it can slow down a bit after getting hit can make it more controllable.

**Comment by DinosaurPower** [comment]
Hi Max! Sorry for late feedback!
First of all, I totally love the theme combination and the physics of the ball. They have this _feel_ that make game dynamic but also realistic in their response.
On the other hand, I've noticed few issues.
First of all, I would add keyboard controls so that it would be accessible both from mobile device and compewter just for easier playtesting. 
ANother thing I would add is background for menu as the dark text is hard to read on dark background.
Additionally, I wanted to notice your work on the different calculations (as like score and amount of games played), it gives game some extra feedback towards player, what adds this extra serotonin to the overall experience. 
Good job!!! Looking forward the final version!!

**Comment by redmoe** [comment]
To create a gif, use a screen recorder such as [LICEcap](https://www.cockos.com/licecap/). Set the window size to be around Unity game view and then hit record. There should be an option to "Automatically Stop After" X number of seconds, consider setting it to about 5.

You can then switch to Unity and play the game while recording.

**Review by pecom291 — COMMENTED** [review]
I actually used a combination of knowledge from a friend in terms of coding to help put this part together, he also helped greatly in helping organize the code. I did however primarily reference some code from previous 2D games I've made from last year and semester!

**Review by redmoe — APPROVED** [review]

**Pull request merged**